File: backend-mercado-libre/models/productocarrito.js
const { Model } = require('sequelize');

module.exports = (sequelize, DataTypes) => {
  class productocarrito extends Model {
    static associate(models) {
      productocarrito.belongsTo(models.carrito, { foreignKey: 'idcarrito' });
      productocarrito.belongsTo(models.producto, { foreignKey: 'idproducto', as: 'producto' });
    }
  }

  productocarrito.init({
    idcarrito: {
      type: DataTypes.UUID,
      allowNull: false,
      primaryKey: true,
      references: { model: 'carritos', key: 'id' }
    },
    idproducto: {
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true,
      references: { model: 'productos', key: 'id' }
    },
    cantidad: {
      type: DataTypes.INTEGER,
      allowNull: false
    },
    subtotal: {
      type: DataTypes.DECIMAL(10,2),
      allowNull: false
    }
  }, {
    sequelize,
    modelName: 'productocarrito',
    freezeTableName: true,
    timestamps: true
  });

  return productocarrito;
};

File: frontend-mercado-libre/Models/ProductoCarrito.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace frontendnet.Models;

public class ProductoCarrito
{
    [Required(ErrorMessage = "El campo {0} es obligatorio.")]
    [JsonPropertyName("idcarrito")]
    public required string IdCarrito { get; set; }

    [Required(ErrorMessage = "El campo {0} es obligatorio.")]
    [JsonPropertyName("idproducto")]
    public required int IdProducto { get; set; }

    public int? Cantidad { get; set; }

    public decimal? Subtotal { get; set; }

     public Producto? Producto { get; set; }
}
